loopchain/peer/block_manager.py

```diff
@@ -811,6 +821,21 @@ def vote_unconfirmed_block(self, block_hash, is_validated):
                           f"height({unconfirmed_block.header.height}) "
                           f"unconfirmed_block({unconfirmed_block.header.hash.hex()})")
 
-        await self._vote(unconfirmed_block)
-        self.__channel_service.reset_leader(unconfirmed_block.header.next_leader.hex_hx())
+        try:
+            self.__validate_unconfirmed_block_height(unconfirmed_block)
+        except BlockchainError as e:
+            util.logger.debug(e)
+            return
+
+        try:
+            self.add_unconfirmed_block(unconfirmed_block)
+        except AddUnconfirmedBlock as e:
```

Comment:
What are differences between `AddUnconfirmedBlock` and `AlreadyAddedUnconfirmedBlock`?

Reply:
AlreadyAddedUnconfirmedBlock that has been changed name to DuplicationUnconfirmedBlock means that the unconfirmed block is just already received.
AddUnconfirmedBlock changed that has been changed name to InvalidUnconfirmedBlock is really error.